src/Uno.UI/UI/Xaml/Controls/TextBox/SinglelineTextBoxDelegate.iOS.cs

```diff
@@ -39,8 +39,16 @@ public override bool ShouldChangeCharacters(UITextField textField, NSRange range
 				{
 					var newLength = textBoxView.Text.Length + replacementString.Length - range.Length;
 					return newLength <= _textBox.GetTarget()?.MaxLength;
-				};
-			}
+				};
+
+				if (_textBox.GetTarget() is not TextBox textBox)
+				{
+					return false;
+				}

```

Comment:
Minor suggest, you could move this condition up (line 32) and replace the existing `_textBox.GetTarget()?.` with `textBox.`